charts/traces/values.yaml

```diff
@@ -119,17 +114,13 @@ opentelemetry-collector:
       probabilistic_sampler:
         hash_seed: 22
         sampling_percentage: 100
-      resource:
-        attributes:
-          - key: k8s.cluster.uid
-            value: "${OBSERVE_CLUSTER}"
-            action: insert
       k8sattributes:
         passthrough: false
         extract:
           metadata:
             - k8s.pod.name
             - k8s.namespace.name
+            - k8s.cluster.uid
```

Comment:
Do we have the necessary cluster role privileges? How does the processor retrieve the UID?

Reply:
Based on what I read [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.88.0/processor/k8sattributesprocessor#role-based-access-control), we would need an additional resource (namespaces) in RBAC. There's a comprehensive example [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.88.0/processor/k8sattributesprocessor/testdata/e2e/collector/configmap.yaml) which also refers to a similar RBAC setup. We don't collect deployment / replicaset attributes so we don't need any more than this.

Additionally, our config was not [filtering](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/v0.88.0/processor/k8sattributesprocessor#as-an-agent) down to the node the daemonset was running in, without which it would try and list every pod across every namespace. I also added that, but let me know if there's a reason for the way it was.

Reply:
I assume if you need to grab the UID of `kube-system` you will need a cluster role and cluster role binding. I believe we already have that set up as part of the service account because it was required by cluster-info, but we should make sure that is the case and remove the cluster info dependency entirely.

As for filtering, we do not assume otel-collector is run as a daemonset. There are cases where that is not an option (e.g. EKS on Fargate)

Reply:
Yes, cluster-info dependency is no longer needed. I did note that we need permissions on `namespaces` in addition to `pods` for the new resource detection to work correctly so I added that.

Also, the Helm defaults to a daemonset, which is why I added the filtering. 
https://github.com/observeinc/helm-charts/blob/7b5df82f49bc2566149a28b48291305c3bc4a321/charts/traces/values.yaml#L13